### PR TITLE
Add sphinx documentation and publish to ReadTheDocs

### LINKED ISSUE #5: Add sphinx documentation and publish to ReadTheDocs
Pretty simple feature.

- [x] Include a `docs/source` folder containing documentation.
- [x] Add `sphinx-build` GitHub actions
- [x] `git_info` update
     - [x] Static type hinting in `git_info`
     - [x] Docstrings update
- [x] Rename to `redata-commons`
- [ ] ~For simplicity, I recommend changing README.md to README.rst so that all documentation is using consistent format.~ (this is a separate issue that will be addressed after publishing)
- [x] Publish it as redata-commons.readthedocs.org

### PR DESCRIPTION
**Pull request by astrochun**
Closes #5

ToDo:
- [x] Bump version: v0.1.0 -> v0.2.0
- [x] Update CHANGELOG.md

Commit history:
- Update gitignore for docs/build [ci skip]
- Add sphinx build files
- Add docs/source rst files
- Adjust python-package to skip docs
- git_info: Static type hinting
- Fix unit test for required input_path in GitInfo
- Add sphinx-build GitHub Actions
- Remove overview.rst (not needed) [docs only]

Extra info:

- extlinks [documentation](https://www.sphinx-doc.org/en/master/usage/extensions/extlinks.html)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -5,12 +5,12 @@
```
Comment (damian-romero) [inline setup.py:5]: I am not sure if the following is needed for people who want to install from source (meaning `git clone`, etc.):
```suggestion
with open("docs/requirements.txt", "r") as fr:
    requirements = fr.read().splitlines()
```
and then...
```python
    setup(... etc
    install_requires=requirements
```

**Review by damian-romero — COMMENTED** [review]
The documentation works well and I did not make any changes. 

A few thoughts:

- Do we want to have any one of the ReDATA logos on the front page of the RTDs?
- I am getting the following warning after running`make html`: `WARNING: html_static_path entry '_static' does not exist`. This can be removed easily (see this [issue](https://github.com/readthedocs/readthedocs.org/issues/1776#issuecomment-149684640)), but I did not suggest it since I am not sure if we want `html_static_path = [_static]` or not. 
- I left some comments that don't have to do with documentation. I did not implement them because I wasn't sure if those would be changes we wanted to make or not.

File: setup.py (hunk 1 above)
Comment (astrochun) [reply setup.py:5]: I'm not entirely sure if we should. First, by adding this to `setup.py` it adds the `sphinx` libraries and dependencies for any install. If the intent is to just use the code by doing `python setup.py install` then you don't need sphinx, etc. We always prefer to be on the side of simpler build. I think there is a way to do this more elegantly, but not entirely sure. For example, tests dependencies can be specified via the `tests_require` argument. But there's no option for `setuptools`. See: https://github.com/pypa/setuptools/issues/1820

Note that the `sphinx` CI builds uses the `docs/` folder, which is why there is a separate requirements.txt there. It was the simplest way to ensure the `sphinx` builds pass.

**Review by astrochun — COMMENTED** [review]
Thanks for review. I think this looks good otherwise. See response to your suggestion

**Pull request merged**